lib/Doctrine/ORM/EntityRepository.php

```diff
@@ -302,7 +310,7 @@ protected function getClassMetadata()
      * return a new collection containing these elements.
      *
      * @return LazyCriteriaCollection
-     * @psalm-return Collection<int, T>
+     * @psalm-return LazyCriteriaCollection<int, T>
```

Comment:
Did you chose to replace an interface with an implementation to drop a dependency to the Collections project? Or also because of `Selectable`?

Reply:
I wanted to match the existing `@return` annotation.

Reply:
I'd prefer to use an interface instead of an implementation detail, but I see that phpdoc already lists `LazyCriteriaCollection` above. I'm not sure if it can be set to an interface in 2.12.

Reply:
I'm fine either way. I could also change this to an intersection of `AbstractLazyCollection` and `Selectable` which is probably the contract that we want to expose here if we consider `LazyCriteriaCollection` to be an implementation detail.

Reply:
I think that's a compromise.

Reply:
Done.